packages/react/slot/src/Slot.tsx

```diff
@@ -121,6 +123,30 @@ function mergeProps(slotProps: AnyProps, childProps: AnyProps) {
   return { ...slotProps, ...overrideProps };
 }
 
+// Before React 19 accessing `element.props.ref` will throw a warning and suggest using `element.ref`
+// After React 19 accessing `element.ref` does the opposite.
+// https://github.com/facebook/react/pull/28348
+//
+// Access the ref using the method that doesn't yield a warning.
+function getElementRef(element: React.ReactElement) {
```

Comment:
Same method as before, let's extract to a utility.

Reply:
It's quite heavy for the bundle size. Did you consider:

```jsx
return React.version.split('.')[0] >= '19' ? element.props.ref : element.ref;
```

or

```jsx
return children.props.propertyIsEnumerable('ref') ? element.props.ref : element.ref;
```

Reply:
If no ref is set on the element, element.props.propertyIsEnumerable('ref') will return `false`, making the program access `element.ref` that will trigger a warning, no?